LayoutTests/fast/files/blob-formdata-cancel.html

```diff
@@ -0,0 +1,28 @@
+<!doctype html>
+<html>
+    <head>
+        <meta charset="utf-8">
+        <script src="../../resources/testharness.js"></script>
+        <script src="../../resources/testharnessreport.js"></script>
+        <script src="../../resources/gc.js"></script>
+    </head>
+    <body>
+        <script>
+function textFromBlobFormData()
+{
+    const blob = new Blob(["123"], {type: "text/plain"});
+    const formData = new FormData();
+    formData.append("file", blob, "");
+    const response = new Response(formData);
+    response.text();
```

Comment:
This test shows a bug. We're not supposed to garbage collect the response while there is a `text()` request pending.

Just tweak this line to do:
```
response.text().then((r) => {
    console.log("Read");
});
```

You'll see "Read" printed 10 times in Chrome, as expected. However, in Safari, you'll get a much smaller number of logging because we're collecting the response too eagerly. This is actually a pretty bad bug.

Reply:
True, I'll fix it.
This is not really related to this PR which is about doing less work in destructors.

Reply:
Feels related since we shouldn't be destroying the BlobLoader in the middle of a load anymore?

Reply:
Honestly, this PR looks a little suspect to me. Seems odd for the BlobLoader to pass null to the completion handler on cancellation.

Reply:
Also, the commit log says that ref-ing on object during destruction isn't safe but I don't think this is true. This is why we have `RefAllowingPartiallyDestroyed` / `RefPtrAllowingPartiallyDestroyed`

Reply:
In general, passing an object that is being destroyed in a lambda seems like an anti-pattern to me and can lead to safety issues.

Our loading code is a good example: when we cancel a load, we do not get a callback stating that the load is cancelled. The expectation is that all callbacks are neutered given I lost interest in the load.

The alternative to passing something other than a being destroyed loader (like nullptr) is to not execute the callback at all. I'll update the PR accordingly.

> Feels related since we shouldn't be destroying the BlobLoader in the middle of a load anymore?

This is true for the added test but there are other places where it is valid to destroy the blob loader in the middle of a load, like NetworkSendQueue.
NetworkSendQueue::clear() comment is an example of special handling we would avoid by not calling BlobLoader callback when cancelling it.